### ISSUE TITLE
[E] improve OssnThemes->upload()

### ISSUE THREAD

**Issue by githubertus** [opensource-socialnetwork/opensource-socialnetwork#1694]
with better error logs as already done in OssnComponents->upload()

(the current one and only error message for any kind of issue isn't really helpful)

**Comment by lianglee** [comment]
Besides those callback enable, disable, i think these are useful only for different component then the actual component to track down actions,  because , just example delete callback can not be used for the component that is actually being deleted.  So We need something to call a standalone file on delete , enable, deactivate for example

components/$com->id/<enable/disable/deactivate file>

**Comment by lianglee** [comment]
Now you can add 

enable.php
disable.php
delete.php 

these files will be called only once when the action on component is performed,  theme didn't have disabled.php

**Comment by lianglee** [comment]
enable.php called before enabled [COM|THEME]
disable.php called before disabled  [COM]
delete.php  called before deleted [COM|THEME]

enabled.php called after enabled [COM|THEME]
disabled.php called after disabled [COM]
deleted.php called after deleted [COM|THEME]
